src/docfx/build/xref/XrefResolver.cs

```diff
@@ -320,7 +346,8 @@ private static string RemoveSharingHost(string url, string hostName)
         if (_externalXrefMap.Value.TryGetValue(uid, out var spec))
         {
             var href = RemoveSharingHost(spec!.Href, _config.HostName);
-            return (spec, href);
+
+            return (spec, HostNameUtility.ReplaceHostForUrl(href, _reversedHostNameMap));
```

Comment:
I don't think we need to remove again as line 348 is without host name.

Reply:
It is just a piece of defensive code for the extremely corner case as the following:
the _config.HostName is wrong by mistake like wrong.doc.com ( because we will support overrite hostname from environment var)
but the xref url is start with new.doc.com

Reply:
Such cases should be prevented by test cases. I will not suggest we pollute production codes just to handle such cases.